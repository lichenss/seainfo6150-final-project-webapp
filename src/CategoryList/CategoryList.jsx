import React from 'react'
import styles from './CategoryList.module.css'
import computerscienceImg from '../images/computer_science.jpg';
import businessImg from '../images/business.jpg';
import datascienceImg from '../images/data_science.jpg';
import artsandhumanitiesImg from '../images/arts_and_humanities.jpg';
import physicalscienceandengineeringImg from '../images/physical_science_and_engineering.jpg';
import mathandlogicImg from '../images/math_and_logic.jpg';
import CategoryListItem from '../CategoryListItem/CategoryListItem';

const categories = [
    {
        slug: 0,
        categoryName: 'Computer Science',
        categoryLink:'computer-science',
        imageURI: computerscienceImg,
    },
    {
        slug: 1,
        categoryName: 'Business',
        categoryLink:'business',
        imageURI: businessImg,
    },
    {
        slug: 2,
        categoryName: 'Data Science',
        categoryLink:'data-science',
        imageURI: datascienceImg,
    },

    {
        slug: 3,
        categoryName: 'Arts and Humanities',
        categoryLink:'arts-and-humanities',
        imageURI: artsandhumanitiesImg,
    },
    {
        slug: 4,
        categoryName: 'Physical Science and Engineering',
        categoryLink:'physical-science-and-engineering',
        imageURI: physicalscienceandengineeringImg,
    },
    {
        slug: 5,
        categoryName: 'Math and Logic',
        categoryLink:'math-and-logic',
        imageURI: mathandlogicImg,
    },
];
const CategoryList = () => {
    return (
        <div className={styles.section}>
        <ul className={styles.container}>
            {categories.map((category) => (
                <CategoryListItem category={category} key={category.slug} />
            ))}
        </ul>
        </div>
    )
}
export default CategoryList