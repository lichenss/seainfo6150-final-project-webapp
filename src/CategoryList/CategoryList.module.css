.section{
  background-color: rgb(229, 231, 233);
  margin-top: 70px;
}
.container {
  margin-top: 70px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  
}

@media screen and (min-width: 700px) {
  .categoryContainer {
    grid-template-columns: repeat(2, 1fr);
  }
}

@media screen and (min-width: 1150px) {
  .categoryContainer {
    grid-template-columns: repeat(3, 1fr);
  }
}
