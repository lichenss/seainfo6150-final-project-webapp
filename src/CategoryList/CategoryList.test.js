import React from "react";
import CategoryList from "./CategoryList";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
describe("CategoryList tests", () => {
    it("renders correctly", () => {
      const { container } = render(
      <BrowserRouter>
      <CategoryList />
      </BrowserRouter>);
      expect(container).toMatchSnapshot();
    });
  });