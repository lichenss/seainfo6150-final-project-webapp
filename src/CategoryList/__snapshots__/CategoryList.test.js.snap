// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList tests renders correctly 1`] = `
<div>
  <div
    class="section"
  >
    <ul
      class="container"
    >
      <li
        class="category"
      >
        <a
          href="/category/computer-science"
        >
          <div
            class="imageBlock"
          >
            <img
              alt="Computer Science"
              class="image"
              src="computer_science.jpg"
            />
          </div>
          <h3
            class="title"
          >
            Computer Science
          </h3>
        </a>
      </li>
      <li
        class="category"
      >
        <a
          href="/category/business"
        >
          <div
            class="imageBlock"
          >
            <img
              alt="Business"
              class="image"
              src="business.jpg"
            />
          </div>
          <h3
            class="title"
          >
            Business
          </h3>
        </a>
      </li>
      <li
        class="category"
      >
        <a
          href="/category/data-science"
        >
          <div
            class="imageBlock"
          >
            <img
              alt="Data Science"
              class="image"
              src="data_science.jpg"
            />
          </div>
          <h3
            class="title"
          >
            Data Science
          </h3>
        </a>
      </li>
      <li
        class="category"
      >
        <a
          href="/category/arts-and-humanities"
        >
          <div
            class="imageBlock"
          >
            <img
              alt="Arts and Humanities"
              class="image"
              src="arts_and_humanities.jpg"
            />
          </div>
          <h3
            class="title"
          >
            Arts and Humanities
          </h3>
        </a>
      </li>
      <li
        class="category"
      >
        <a
          href="/category/physical-science-and-engineering"
        >
          <div
            class="imageBlock"
          >
            <img
              alt="Physical Science and Engineering"
              class="image"
              src="physical_science_and_engineering.jpg"
            />
          </div>
          <h3
            class="title"
          >
            Physical Science and Engineering
          </h3>
        </a>
      </li>
      <li
        class="category"
      >
        <a
          href="/category/math-and-logic"
        >
          <div
            class="imageBlock"
          >
            <img
              alt="Math and Logic"
              class="image"
              src="math_and_logic.jpg"
            />
          </div>
          <h3
            class="title"
          >
            Math and Logic
          </h3>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
